### ISSUE TITLE
Redirection not working on emulator

### ISSUE THREAD

**Issue by 0x5d** [FormidableLabs/react-native-app-auth#297]
## Issue

An emulator, upon successfully signing in, the user is never redirected to the app, *unless* the custom tab is being debugged remotely in Chrome's dev tools.
On actual devices it works fine, so I'm not sure to call this a bug, but it might be a symptom of something else not working properly. 

I also wonder if it's related to this: https://github.com/okta/okta-sdk-appauth-android/issues/8

I have set 
```groovy
manifestPlaceholders = [
    appAuthRedirectScheme: 'ticketapp'
]
```
as well as an intent filter for `.MainActivity`:
```xml
      <activity
        android:name=".MainActivity"
        android:label="@string/app_name"
        android:configChanges="keyboard|keyboardHidden|orientation|screenSize"
        android:windowSoftInputMode="adjustResize">
        <intent-filter>
            <action android:name="android.intent.action.MAIN" />
            <category android:name="android.intent.category.LAUNCHER" />
        </intent-filter>
        <intent-filter>
            <action android:name="android.intent.action.VIEW"/>
            <category android:name="android.intent.category.DEFAULT"/>
            <category android:name="android.intent.category.BROWSABLE"/>
            <data android:scheme="ticketapp"/>
        </intent-filter>
      </activity>
```
---

## Environment

- **Your Identity Provider**: `e.g. IdentityServer 4 / Okta / Azure`
- **Platform that you're experiencing the issue on**:
  - [ ] iOS
  - [ ] Android
  - [ ] **iOS** but have not tested behavior on Android
  - [x ] **Android** but have not tested behavior on iOS
  - [ ] Both
- **Are you using Expo, e.g. `ExpoKit`?**
  - [x] No
  - [ ] Yes, I've _not_ ejected
  - [ ] Yes, but I **have** ejected to `ExpoKit`
  - [ ] Yes, but I **have** ejected to vanilla React Native

**Comment by spsaucier** [comment]
I've run into this intermittently as well. Periodically, the app is also not creating the initial authorize webview either, particularly if the app has been open for a while.

**Comment by kadikraman** [comment]
Hi @castillobgr 👋 - what version of the library and React Native are you using?

**Comment by castillobgr** [comment]
Hi @kadikraman ! 
```
    "react-native": "0.59.4",
    "react-native-app-auth": "4.3.1",
```

**Comment by kadikraman** [comment]
Hmm - that's a very strange error, and I can't say I've ever come across it myself. Is it reproducible on another device/emulator?

**Comment by AGIsmail** [comment]
I faced this issue on the Pixel 3 emulator running Android 9.0 and authenticating with 2 different services (1 self-hosted and 1 online).

When authorizing from the browser on an emulator, the following may occur. 
1- the browser attempts to download a file and errors out.
2- the browser returns to the app and reports a data is null error.

I haven't able to get the browser/app to report if there's a problem with the redirectUrl when running on an emulator. The opposite is true when testing on a physical device (i.e, problems do get reported).

If anyone faces this issue, this is how I got beyond it on the emulator:
Run Wipe Data on the emulator from the AVD manager and re-install the app. It should work now, assuming that your redirectUrl is actually configured properly everywhere (e.g., `app/build.gradle` file, `config` object, service provider..).

Configuring the `intent-filter` was unnecessary.
